pkg/api/v1/clients/kube/resource_client.go

```diff
@@ -457,8 +459,28 @@ func (rc *ResourceClient) convertCrdToResource(resourceCrd *v1.Resource) (resour
 				*status = typedStatus
 				return nil
 			}
-			if err := resources.UpdateStatus(withStatus, updateFunc); err != nil {
-				return nil, err
+			updateReporterStatusFunc := func(status *core.ReporterStatus) error {
+				if status == nil {
+					return nil
+				}
+				typedStatus := core.ReporterStatus{}
+				if err := protoutils.UnmarshalMapToProto(resourceCrd.Status, &typedStatus); err != nil {
+					return err
+				}
+				*status = typedStatus
+				return nil
+			}
+			// First attempt to unmarshal ReporterStatus
+			if reporterStatusErr := resources.UpdateReporterStatus(withStatus, updateReporterStatusFunc); reporterStatusErr != nil {
+				// If unmarshalling ReporterStatus failed, the resource likely has a Status instead.
```

Comment:
Can we implement a check on the error message to make sure that the error is what we expect here (the status is of type `Status` rather than `ReporterStatus`), and not another k8s error?

Reply:
I see down below you report both errors if unmarshalling to `Status` fails. This isn't necessary then, but a nit. We might want to just catch and return the error earlier on if it's not an `Unmarshal` error.

Reply:
I like this idea, but not sure if there's a great way I can catch the specific error. For example, the error returned when attempting to unmarshal a `ReporterStatus` into a `Status` looks like: `unknown field "statuses" in core.solo.io.Status` of the type `*errors.errorString`.  Who knows if there are other unmarshalling errors that could be returned with a different message format?

If there was a `UnmarshalError` error type that was returned, it would be much easier to just look for that error type rather than trying to match a string from a generic error